### PR TITLE
Provide secret scanning for local setup

### PR DESCRIPTION
**Pull request by serverhorror**
Description here ...

Closes #...
Fixes #...

Tasks: 
- [x] Updated documentation in `docs/modules/...` directory
- [ ] Ran tests in `<quickstarter>/testdata` directory

### CODE REVIEW

File: docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+== Secret scanning with gitleaks
+
+This repository has a pre-commit hook that runs gitleaks on every commit.
+To install the hook, run:
+
+```
+pip install pre-commit
```
Comment (renedupont) [inline docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc:7]: Does that mean I would need to install python+pip on my machine to use this?

Comment (fundacioemys) [reply docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc:7]: consider also there are other ways to install it locally -> https://pre-commit.com/#install

Comment (gerardcl) [reply docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc:7]: consider also there are other ways to install it locally -> https://pre-commit.com/#install

**Review by gerardcl — DISMISSED** [review]
LGTM! 🚀

File: docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc (hunk 1 above)
Comment (renedupont) [reply docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc:7]: > consider also there are other ways to install it locally -> https://pre-commit.com/#install

Do you mean by using homebrew or conda? One of them would need to be installed first as well to install pre-commit. 

But yes, the question is answered for me, thx!

File: docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+== Secret scanning with gitleaks
+
+This repository has a pre-commit hook that runs gitleaks on every commit.
+To install the hook, run:
```
Comment (cschweikert) [inline docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc:4]: I'd suggest to include a reference to the alternatives, e.g.
```suggestion
To https://pre-commit.com/#install[install the hook], run:
```
This would lower the barrier of actually using a pre-commit hook.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ### Added
+- Added scret scanning in docker plain ([#963](https://github.com/opendevstack/ods-quickstarters/pull/963))
```
Comment (cschweikert) [inline CHANGELOG.md:4]:
```suggestion
- Added secret scanning in docker plain ([#963](https://github.com/opendevstack/ods-quickstarters/pull/963))
```

**Review by cschweikert — COMMENTED** [review]
Good stuff! Thanks for introducing this 😊

File: docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc (hunk 1 above)
Comment (henrjk) [inline docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc:7]: Let's rather link to the page Gerard already mentioned indicating that there are also 0-dependency and platform installation avenues. It looks like https://pre-commit.com is blocked internally at the moment however.

File: be-fe-mono-repo-plain/files/.pre-commit-config.yaml
Comment (stitakis) [inline be-fe-mono-repo-plain/files/.pre-commit-config.yaml:1]: The proposed pre-commit configuration looks good. However it relies on the default configuration of gitleaks.
What about if have an urgent need to provide custom rules? 
Following configuration would allows us to do it by extending the default configuration with custom configuration from opendevstack. 

```
repos:
  - repo: local
    hooks:
      - id: curl
        name: Download opendevstack additional gitleaks configuration (extends default gitleaks rules)
        entry: curl https://raw.githubusercontent.com/opendevstack/pre-commit-hooks/opendevstack-custom-gitleaks-config.toml -o .opendevstack-gitleaks.toml
        language: system
  - repo: https://github.com/gitleaks/gitleaks
    rev: v8.16.1
    hooks:
      - id: gitleaks
        args: [--source, .opendevstack-gitleaks.toml]

```

To be able to extend the default gitleak rules the custom configuration would need to start with:
```
[extend]
useDefault = true

[<custom-rules-from-here>]
...
```

I'm providing the above example for discussion. All details are not freshed out yet.  
In summary, this approach adds a bit more complexity to the configuration, but would allow us to provide custom gitleaks rules and this way closer to the reality of enterprise needs.

File: docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc (hunk 2 above)
Comment (tbugfinder) [reply docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc:4]: Cloud quickstarters include Makefile to ease local setup.
Maybe we could add Makefile in any qs and just harmonize on `make init`.

https://github.com/opendevstack/ods-quickstarters/blob/master/inf-terraform-aws/files/Makefile#L36

File: be-fe-mono-repo-plain/files/.pre-commit-config.yaml
Comment (stitakis) [reply be-fe-mono-repo-plain/files/.pre-commit-config.yaml:1]: From my point of view, this doesn't collide with review process, as long the main gitleaks configuration is hosted in opendevstack.
At the end of the day, all of this works on the development environment locally and only if the user install the tools.
To adopt the proposed example there is a bit work to complete, like improving the CURL command an also, pushing the first version of the main gitleaks configuration (we need to decide where to host it) and testing it.
I suggest we merge this pull request as it is and create new ones to implement the proposal.
Sounds good?

File: docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc (hunk 2 above)
Comment (tbugfinder) [reply docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc]: It doesn't have to be make itself  but the same approach, if task/Taskfile is better....

Though, not sure how many developers use Windows for all those QS environments and if those would not use WSL2 anyway.

File: be-fe-mono-repo-plain/files/.pre-commit-config.yaml
Comment (stitakis) [reply be-fe-mono-repo-plain/files/.pre-commit-config.yaml:1]: not sure, but we will figure it out, anyway this PR looks good to me

**Review by cschweikert — APPROVED** [review]

**Review by henrjk — APPROVED** [review]

**Review by stitakis — APPROVED** [review]

File: docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc (hunk 2 above)
Comment (nichtraunzer) [reply docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc]: the Cloud QS is heavily using make and we confuse users when we introduce a different approach here.

Comment (renedupont) [reply docs/modules/quickstarters/partials/secret-scanning-with-gitleaks.adoc]: Other quickstarters are using different build tools, or is this PR somehow specific to the cloud qs?

**Review by michaelsauter — APPROVED** [review]
Sorry wasn't aware you're waiting for approval from everyone :) This is great, please go ahead!

**Review by renedupont — APPROVED** [review]

**Review by oalyman — APPROVED** [review]

**Review by bljubisic — APPROVED** [review]
